src/main/constants.ts

```diff
@@ -6,10 +6,16 @@
 
 import { app } from "electron";
 import { existsSync, readdirSync, renameSync, rmdirSync } from "fs";
-import { join } from "path";
+import { dirname, join } from "path";
+
+const vesktopDir = dirname(process.execPath);
```

Comment:
this already looks a lot better, but i think when running unpackaged (eg via pnpm start), this will correspond to the electron binary which will lead to it thinking its portable

which technically it is but it would probably be better to treat it as normal? might be as simple as checking if process.execPath is "electron"